integration/dkg/dkg_emulator_suite.go

```diff
@@ -398,14 +417,17 @@ func (s *DKGSuite) initEngines(node *node, ids flow.IdentityList) {
 	)
 	require.NoError(s.T(), err)
 
-	// We add a hook to the logger such that the test fails if the broker writes
-	// a Warn log, which happens when it flags or disqualifies a node
-	hook := zerolog.HookFunc(func(e *zerolog.Event, level zerolog.Level, message string) {
-		if level == zerolog.WarnLevel {
-			s.T().Fatal("DKG flagging misbehaviour")
-		}
-	})
-	controllerFactoryLogger := zerolog.New(os.Stdout).Hook(hook)
+	controllerFactoryLogger := zerolog.New(os.Stdout)
+	if s.checkDKGUnhappy {
+		// We add a hook to the logger such that the test fails if the broker writes
+		// a Warn log, which happens when it flags or disqualifies a node
+		hook := zerolog.HookFunc(func(e *zerolog.Event, level zerolog.Level, message string) {
```

Comment:
AFAICT:
- the only implementation of the [`DKGProcessor` interface](https://github.com/onflow/flow-go/blob/d648e6a1d0e60f0f9e104c4592f80a8f041db16c/crypto/dkg.go#L139), which does commit the DKG to log upon flagging / disqualification, is [in dkg_test](https://github.com/onflow/flow-go/blob/d648e6a1d0e60f0f9e104c4592f80a8f041db16c/crypto/dkg_test.go#L440-L509),
- this is fine for the purposed of this test, but [that implementation logs disqualification / flags at the Info level](https://github.com/onflow/flow-go/blob/d648e6a1d0e60f0f9e104c4592f80a8f041db16c/crypto/dkg_test.go#L487-L495), which is admittedly more weird.


=> this may not work as intended, or did I miss something?

Reply:
The [DKGBroker](https://github.com/onflow/flow-go/blob/martin/5440-dkg-unhappy-path/module/dkg/broker.go) implements the DKGProcessor interface. This is what's being tested here.

Reply:
Non-contributory behaviour is tested in the crypto package directly. The goal of these integrations tests is to check if anything is breaking at another layer ( network, engines, broker, ... etc)

Reply:
> The DKGBroker implements the DKGProcessor interface. This is what's being tested here.

🤦  yep, you're on the epochs branch, of course.